src/Orleans/Core/GrainStateStorageBridge.cs

```diff
@@ -12,10 +12,11 @@ namespace Orleans.Core
     internal class GrainStateStorageBridge : IStorage
     {
         private readonly IStorageProvider store;
-        private readonly Grain grain;
+        private readonly IStatefulGrain grain;
```

Comment:
can you please explain what is the difference between `grain` and `baseGrain` here and why we need both?